server/src/main/java/org/apache/kafka/server/share/fetch/PartitionMaxBytesStrategy.java

```diff
@@ -58,7 +64,27 @@ static PartitionMaxBytesStrategy type(StrategyType type) {
     private static LinkedHashMap<TopicIdPartition, Integer> uniformPartitionMaxBytes(int requestMaxBytes, Set<TopicIdPartition> partitions, int acquiredPartitionsSize) {
         checkValidArguments(requestMaxBytes, partitions, acquiredPartitionsSize);
         LinkedHashMap<TopicIdPartition, Integer> partitionMaxBytes = new LinkedHashMap<>();
-        partitions.forEach(partition -> partitionMaxBytes.put(partition, requestMaxBytes / acquiredPartitionsSize));
+        if (requestMaxBytes >= acquiredPartitionsSize) {
+            // Case 1: requestMaxBytes can be evenly distributed within partitions.
+            partitions.forEach(partition -> partitionMaxBytes.put(partition, requestMaxBytes / acquiredPartitionsSize));
```

Comment:
Even for this case will the logic work always i.e. say request bytes are 10 and there are 6 acquired topic partitions then each partition maxbytes will be set to 1. Meaning we will be left with 4 bytes, should we not factor that in as well?